### DISCUSSION TITLE
Unable to activate ST7789 on my DSTIKE ESP8266

### DISCUSSION THREAD

**Discussion by marsrakete** [Bodmer/TFT_eSPI#1615] [General]
Using Read_User_Setup I get this:

`TFT_eSPI ver = 2.4.32`
`Processor    = ESP8266`
`Frequency    = 160MHz`
`Voltage      = 3.33V`
`Transactions = No`
`Interface    = SPI`
`SPI overlap  = No`
`Display driver = 7789`
`Display width  = 240`
`Display height = 240`
`MOSI    = GPIO 2`
`SCK     = GPIO 1`
`TFT_DC   = PIN_D3`
`TFT_RST  = PIN_D4`
`Font GLCD   loaded`
`Font 2      loaded`
`Font 4      loaded`
`Font 6      loaded`
`Font 7      loaded`
`Font 8      loaded`
`Smooth font enabled`
`Display SPI frequency = 27.00`

My User_Setup.h contains this:
`#define ST7789_DRIVER`
`#define TFT_WIDTH  240`
`#define TFT_HEIGHT 240`
`#define TFT_CS    -1`
`#define TFT_DC    0`
`#define TFT_RST   2`
`#define TFT_MOSI  4`
`#define TFT_SCLK  5`

I am using the Arduino version 1.8.13
The library TFT_eSPI installed is version 2.4.32

Board is this one:
https://dstike.com/products/datike-d-duino-v6   
The page lists the GPIO-Ports as defined in my User_Setup.
However Read_User_Setup does not show the right ports. 
None of the examples of TFT_eSPI works. The Adafruit-Examples do work (but slow).

Why does Read_User_Setup lists TFT_MOSI as GPIO2 whereas my User_Setup is Pin 4?
What am I missing? I can only guess, that my setup is completely wrong.

Any help welcome.

**Comment by Bodmer** [comment]
The sketch assumes a NodeMCU ESP8266 board has been used with pins specified in the PIN_Dx format:
#define TFT_DC   PIN_D3

Pin [mapping is here](https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup_Select.h#L227) .

**Comment by Bodmer** [comment]
You may need to use a driver 2 variant instead:
#define ST7789_2_DRIVER

**Comment by marsrakete** [comment]
I have seen the mapping in User_Setup_Select, but couldn't find the right setup. 
The DSTIKE board requires this:
Display SDA(MOSI): GPIO 4
Display SCL/SCK: GPIO 5
Display DC: GPIO 0
Display RST: GPIO2
Display Background LED : GPIO16

Then this is what I used in User_setup.h, using the mapping in User_Setup_Select:
//#define TFT_CS    -1
#define TFT_DC    PIN_D3
#define TFT_RST   PIN_D4   
#define TFT_MOSI  PIN_D2
#define TFT_SCLK  PIN_D1 
#define TFT_BL    PIN_D0  


Read_User_Setup reports:
MOSI    = GPIO 2
SCK     = GPIO 1
TFT_DC   = PIN_D3
TFT_RST  = PIN_D4
TFT_BL           = PIN_D0

The mapping still gets wrong somewhere. I tried both, ST7789_DRIVER and ST7789_2_DRIVER.

**Comment by Bodmer** [comment]
The library requires that you use the hardware SPI, so the hardware GPIO pins 5 for SCLK, GPIO 7 for MOSI and optionally GPIO 6 for MISO. You  can choose the DC and CS lines. Note that the use of some pins can hold the ESP8266 in boot or FLASH mode, check online for pin information if you are not sure.

Although any SPI pins ( MOSI, MISO SCLK) can be used with some libraries then software bit bashed SPI will be used which will be very slow.

**Comment by Bodmer** [comment]
Looking at the hardware design it appears the SPI pins are fixed and so cannot be easily changed. This means the TFT _eSPI library will not work with that board.

**Comment by marsrakete** [comment]
Ah, darn. You get what you pay for? The connection from the ESP to the TFT is fixed on the board. I really appreciate your help anyway! I was hoping to get your fast library to work. 

Case closed.
